js/controller/ListController.js

```diff
@@ -53,44 +54,67 @@ app.controller('ListController', function ($scope, $location, $filter, BoardServ
 	};
 	$scope.colors = ['0082c9', '00c9c6','00c906', 'c92b00', 'F1DB50', '7C31CC', '3A3B3D', 'CACBCD'];
 	$scope.boardservice = BoardService;
-	$scope.newBoard.color = calculateNewColor();
 	$scope.updatingBoard = null;
 
-	// FIXME: not nice, but we want to load this only once
-	if($element.attr('id') === 'app-navigation') {
-		BoardService.fetchAll().then(function(data) {
-			$scope.filterData();
-		}, function (error) {
-			// TODO: show error when loading fails
-		});
-	}
-
-	$scope.filterData = function () {
-		angular.copy($scope.boardservice.getData(), $scope.boardservice.sorted);
-		angular.copy($scope.boardservice.sorted, $scope.boardservice.sidebar);
-		$scope.boardservice.sidebar = $filter('orderBy')($scope.boardservice.sidebar, 'title');
-		$scope.boardservice.sidebar = $filter('cardFilter')($scope.boardservice.sidebar, {archived: false});
-
-		if ($scope.status.filter === 'archived') {
-			var filter = {};
-			filter[$scope.status.filter] = true;
-			$scope.boardservice.sorted = $filter('cardFilter')($scope.boardservice.sorted, filter);
-		} else if ($scope.status.filter === 'shared') {
-			$scope.boardservice.sorted = $filter('cardFilter')($scope.boardservice.sorted, {archived: false});
-			$scope.boardservice.sorted = $filter('boardFilterAcl')($scope.boardservice.sorted);
+
+	var finishedLoading = function() {
+		filterData();
+		$scope.newBoard.color = calculateNewColor();
+	};
+
+	var filterData = function () {
+		if($element.attr('id') === 'app-navigation') {
+			$scope.boardservice.sidebar = $scope.boardservice.getData();
+			$scope.boardservice.sidebar = $filter('orderBy')($scope.boardservice.sidebar, 'title');
+			$scope.boardservice.sidebar = $filter('cardFilter')($scope.boardservice.sidebar, {archived: false});
 		} else {
-			$scope.boardservice.sorted = $filter('cardFilter')($scope.boardservice.sorted, {archived: false});
+			$scope.boardservice.sorted = $scope.boardservice.getData();
+			if ($scope.status.filter === 'archived') {
+				var filter = {};
+				filter[$scope.status.filter] = true;
+				$scope.boardservice.sorted = $filter('cardFilter')($scope.boardservice.sorted, filter);
+			} else if ($scope.status.filter === 'shared') {
```

Comment:
Isn't better to use enums instead of string to indicate the filter state? :)

Maybe it can go to a separated issue

Reply:
Hmm, the filter state is set as a state param where we have it as a string anyway. So until we don't use it more often in the code, i guess that is fine for now. But if you want to improve here, feel free to do so. 😉